cmd/lifecycle/analyzer.go

```diff
@@ -225,6 +264,30 @@ func (a *analyzeCmd) restoresLayerMetadata() bool {
 	return !a.platformAPIVersionGreaterThan06()
 }
 
+func (a *analyzeCmd) supportsRunImage() bool {
+	return a.platformAPIVersionGreaterThan06()
+}
+
+func (a *analyzeCmd) validateRunImageInput() error {
+	if !a.supportsRunImage() && a.runImageRef != "" {
+		return errors.New("-run-image is unsupported")
+	}
+	return nil
+}
+
+func (a *analyzeCmd) populateRunImageIfNeeded(stackMD platform.StackMetadata, targetRegistry string) error {
+	if !a.supportsRunImage() || a.runImageRef != "" {
+		return nil
+	}
```

Comment:
I think that adding a function with a meaningful name or even just adding a comment for this if condition will make the code clearer.

Reply:
What would you call this function?

Reply:
I don't remember what my thoughts were when I wrote this comment 🤷‍♀️.
Reading this again, I think that the code is clear enough.